perfkitbenchmarker/virtual_machine.py

```diff
@@ -148,6 +163,17 @@ def __repr__(self):
   def __str__(self):
     return self.ip_address
 
+  def __getstate__(self):
+    """Get state, excluding _has_remote_command_script_lock."""
```

Comment:
Can you add comment here explaining why the return result needs to have the _has_remote_command_script_lock removed from this function before returning to the caller? Can the caller just ignore this field then?